integrations/access/jira/types.go

```diff
@@ -18,11 +18,79 @@
 
 package jira
 
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
 // Jira REST API resources
 
+// ErrorResult is used to parse the errors from Jira.
+// The JSON Schema is specified here:
+// https://docs.atlassian.com/software/jira/docs/api/REST/1000.1223.0/#error-responses
+// However JIRA does not consistently respect the schema (especially for old instances).
+// We need to support legacy errors as well (array of strings).
 type ErrorResult struct {
-	ErrorMessages []string `url:"errorMessages"`
-	Errors        []string `url:"errors"`
+	ErrorMessages []string     `url:"errorMessages"`
+	Details       ErrorDetails `url:"errors"`
+}
+
+// Error implements the error interface and returns a string describing the
+// error returned by Jira.
+func (e ErrorResult) Error() string {
+	sb := strings.Builder{}
+	if len(e.ErrorMessages) > 0 {
+		sb.WriteString(fmt.Sprintf("error messages: %s ", e.ErrorMessages))
```

Comment:
Nit: Should we maybe `string.Join(e.ErrorMessages, ";")` or something like this here?

Reply:
I always find those hard to read when multiple error happen, I'd prefer JSON-style array for those, which is done natively by Sprintf %s.

Reply:
I'm not sure about the error messages format (do they include some punctuation?), but adding a join char should help with readability.
Otherwise we might end up with a long string without knowing where a particular ErrorMessage starts and where it ends
Eg
```
V=[issue invalid format error]
using join=issue invalid; format error
```